tests/mtd_mapper/main.c

```diff
@@ -0,0 +1,254 @@
+    .mtd = &dev,
+    .lock = MUTEX_INIT,
+    .init = false,
+};
+
+static mtd_mapper_region_t _region_a = {
+    .mtd = {
+        .driver = &mtd_mapper_driver,
+        .sector_count = SECTOR_COUNT / 2,
+        .pages_per_sector = PAGE_PER_SECTOR,
+        .page_size = PAGE_SIZE,
+    },
+    .parent = &_parent,
+    .offset = 0,
+};
+
+static mtd_mapper_region_t _region_b = {
+    .mtd = {
+        .driver = &mtd_mapper_driver,
+        .sector_count = SECTOR_COUNT / 2,
+        .pages_per_sector = PAGE_PER_SECTOR,
+        .page_size = PAGE_SIZE,
+    },
+    .parent = &_parent,
+    .offset = PAGE_PER_SECTOR * PAGE_SIZE * SECTOR_COUNT / 2,
+};
+
+static mtd_dev_t *_dev_a = &_region_a.mtd;
+static mtd_dev_t *_dev_b = &_region_b.mtd;
+
+static void _test_mem(uint8_t *buffer, size_t len, uint8_t expected)
+{
+    for (size_t i = 0; i < len; i++) {
+        TEST_ASSERT_EQUAL_INT(expected, buffer[i]);
+    }
+}
+
+static void test_mtd_init(void)
+{
+    int ret = mtd_init(_dev_a);
+
+    TEST_ASSERT_EQUAL_INT(0, ret);
+    ret = mtd_init(_dev_b);
+    TEST_ASSERT_EQUAL_INT(0, ret);
+}
+
+static void test_mtd_erase(void)
+{
+    /* Erase first region */
+    int ret = mtd_erase(_dev_a, 0, REGION_FLASH_SIZE);
+
+    TEST_ASSERT_EQUAL_INT(0, ret);
+
+    /* Erase second region */
+    ret = mtd_erase(_dev_b, 0, REGION_FLASH_SIZE);
+    TEST_ASSERT_EQUAL_INT(0, ret);
+
+    ret = mtd_erase(_dev_a, REGION_FLASH_SIZE, 1);
+    TEST_ASSERT_EQUAL_INT(-EOVERFLOW, ret);
+
+    ret = mtd_erase(_dev_b, REGION_FLASH_SIZE, 1);
+    TEST_ASSERT_EQUAL_INT(-EOVERFLOW, ret);
+}
+
+static void test_mtd_read(void)
+{
+    for (uint32_t i = 0; i < REGION_FLASH_SIZE; i += PAGE_SIZE) {
+        mtd_read(_dev_a, _buffer, i, PAGE_SIZE);
+        _test_mem(_buffer, PAGE_SIZE, 0xff);
+    }
+
+    for (uint32_t i = 0; i < REGION_FLASH_SIZE; i += PAGE_SIZE) {
+        mtd_read(_dev_b, _buffer, i, PAGE_SIZE);
+        _test_mem(_buffer, PAGE_SIZE, 0xff);
+    }
+}
+
+static void test_mtd_write(void)
+{
+    static const uint8_t test_val_a = 0xAA;
+
+    memset(_buffer, test_val_a, PAGE_SIZE);
+
+    /* Write first region */
+    for (uint32_t i = 0; i < REGION_FLASH_SIZE; i += PAGE_SIZE) {
+        mtd_write(_dev_a, _buffer, i, PAGE_SIZE);
+    }
+
+    /* Check second region, should still be 0xFF */
+    for (uint32_t i = 0; i < REGION_FLASH_SIZE; i += PAGE_SIZE) {
+        mtd_read(_dev_b, _buffer, i, PAGE_SIZE);
+        _test_mem(_buffer, PAGE_SIZE, 0xff);
+    }
+
+    static const uint8_t test_val_b = 0xBB;
+    memset(_buffer, test_val_b, PAGE_SIZE);
+
+    /* Write second region */
+    for (uint32_t i = 0; i < REGION_FLASH_SIZE; i += PAGE_SIZE) {
+        mtd_write(_dev_b, _buffer, i, PAGE_SIZE);
+    }
+
+    /* Check first region, should still be 0xAA */
+    for (uint32_t i = 0; i < REGION_FLASH_SIZE; i += PAGE_SIZE) {
+        mtd_read(_dev_a, _buffer, i, PAGE_SIZE);
+        _test_mem(_buffer, PAGE_SIZE, 0xAA);
+    }
+}
```

Comment:
you should maybe also check that the second region is properly written

Reply:
Yup, fixed